app/scripts/views/mixins/open-webmail-mixin.js

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+/**
+ * A view mixin that helps confirm views show a redirect button
+ * directly from FxA.
+ */
+define(function (require, exports, module) {
+  'use strict';
+
+  var BaseView = require('views/base');
+  var t = BaseView.t;
+  var _ = require ('underscore');
+
+  var WEBMAIL_SERVICES = [
+    {
+      buttonName: t('Open Gmail'),
+      link: 'https://mail.google.com/mail/u/?authuser=',
+      regex: /@gmail\.com$/,
+      webmailType: 'gmail'
+    },
+    {
+      buttonName: t('Open Hotmail'),
+      link: 'https://outlook.live.com/',
+      regex: /@hotmail\.com$/,
+      webmailType: 'hotmail'
+    },
+    {
+      buttonName: t('Open Yahoo'),
+      link: 'https://mail.yahoo.com',
+      regex: /@yahoo\.com$/,
+      webmailType: 'yahoo'
+    },
+    {
+      buttonName: t('Open Outlook'),
+      link: 'https://outlook.live.com/',
+      regex: /@outlook\.com$/,
+      webmailType: 'outlook'
+    },
+  ];
+
+  return {
+    events: {
+      'click #open-webmail': '_webmailTabOpened'
+    },
+
+    addUserInfo: function (providerLink, email) {
+
+      if (this.getWebmailType(email).contains('gmail')){
```

Comment:
Last thing here: we should stay away from `.contains`. can you just do `this.getWebmailType(email) === gmail` here?